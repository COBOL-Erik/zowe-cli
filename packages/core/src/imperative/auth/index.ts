/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

export * from "./builders/AuthCommandBuilder";
export * from "./builders/AuthLoginCommandBuilder";
export * from "./builders/AuthLogoutCommandBuilder";
export * from "./builders/CompleteAuthGroupBuilder";

export * from "./doc/IAuthHandlerApi";

export * from "./handlers/AbstractAuthHandler";
export * from "./handlers/BaseAuthHandler";
