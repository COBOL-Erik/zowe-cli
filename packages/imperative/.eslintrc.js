module.exports = {
    ...require("../../.eslintrc.js"),
    "parserOptions": {
        "ecmaVersion": 12,
        "sourceType": "module",
        "project": [
            "./tsconfig.json",
            "./__tests__/tsconfig.json",
            "./tsconfig-tests.json"
        ]
    }
}