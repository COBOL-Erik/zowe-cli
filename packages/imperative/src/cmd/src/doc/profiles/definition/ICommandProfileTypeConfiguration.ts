/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { ICommandProfileSchema } from "./ICommandProfileSchema";
import { IProfileTypeConfiguration } from "../../../../..";
import { ICommandProfileAuthConfig } from "./ICommandProfileAuthConfig";

/**
 * Cmd packages additions to the profile manager type configuration document. Used by the CliProfileManager. Allows
 * profiles to be built from command arguments. See the "CliProfileManager" for more information.
 * @export
 * @interface ICommandProfileTypeConfiguration
 * @extends {IProfileTypeConfiguration}
 */
export interface ICommandProfileTypeConfiguration extends IProfileTypeConfiguration {
    /**
     * The JSON schema document. The schema document provides a way to enforce the contents of a profile. The schema
     * conforms exactly to the JSON schema specification. You must supply all properties you would like validated
     * for correctness on the schema, except for "dependencies". Dependency schema checking is generated automatically
     * if you populate the dependencies property of this document.
     *
     * @type {ICommandProfileSchema}
     */
    schema: ICommandProfileSchema;

    /**
     * Configuration for authentication services to associate with this profile type.
     */
    authConfig?: ICommandProfileAuthConfig[];
}
