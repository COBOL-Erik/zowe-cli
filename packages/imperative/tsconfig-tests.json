{
  "compilerOptions": {
    "types": [
      "node",
      "jest"
    ],
    "lib": [
      "esnext"
    ],
    "experimentalDecorators": true,
    "target": "es2015",
    "module": "commonjs",
    "noEmit": true,
    "declaration": true,
    "moduleResolution": "node",
    "noImplicitAny": true,
    "outDir": "./lib",
    "preserveConstEnums": true,
    "removeComments": false,
    "pretty": true,
    "sourceMap": true,
    "newLine": "lf"
  },
  "typeRoots": [
    "./node_modules/@types"
  ],
  "include": [
    "**/__tests__/*",
    "**/__tests__/**/*",
    "**/__mocks__/*"
  ],
  "exclude": [
    "lib",
    "node_modules"
  ],
  "files": [
    "../../__types__/wontache.d.ts"
  ],
  "allowJs": true
}
