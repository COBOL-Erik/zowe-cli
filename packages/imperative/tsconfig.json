{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./lib",
    "stripInternal": true
  },
  "include": [
    "./src"
  ]
}