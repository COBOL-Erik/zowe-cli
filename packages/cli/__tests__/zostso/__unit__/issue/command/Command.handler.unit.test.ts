/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

jest.mock("../../../../../../zostso/lib/IssueTso");
import { IssueTso } from "@zowe/zos-tso-for-zowe-sdk";
import { IHandlerParameters, ImperativeError } from "@zowe/imperative";
import * as Command from "../../../../../src/zostso/issue/command/Command.handler";
import { CommandDefinition } from "../../../../../src/zostso/issue/command/Command.definition";
import { StartTsoData } from "../../../__resources__/StartTsoData";
import {
    UNIT_TEST_ZOSMF_PROF_OPTS,
    UNIT_TEST_PROFILES_ZOSMF_TSO,
    UNIT_TEST_TSO_PROF_OPTS,
} from "../../../../../../../__tests__/__src__/mocks/ZosmfProfileMock";
import { mockHandlerParameters } from "@zowe/cli-test-utils";

const DEFAULT_PARAMETERS: IHandlerParameters = mockHandlerParameters({
    arguments: {
        ...UNIT_TEST_ZOSMF_PROF_OPTS,
        ...UNIT_TEST_TSO_PROF_OPTS,
    },
    positionals: ["zos-tso", "issue", "address-space"],
    definition: CommandDefinition,
    profiles: UNIT_TEST_PROFILES_ZOSMF_TSO,
});
let PARAMS_NO_SSM_STATEFUL = DEFAULT_PARAMETERS;
PARAMS_NO_SSM_STATEFUL.positionals = [
    "zos-tso",
    "issue",
    "cmd",
    "TIME",
    "--sf",
    "--no-ssm",
];
PARAMS_NO_SSM_STATEFUL.arguments.stateful = false;
PARAMS_NO_SSM_STATEFUL.arguments.suppressStartupMessages = false;
PARAMS_NO_SSM_STATEFUL.arguments.commandText = "TIME";

describe("issue command handler tests", () => {
    afterEach(() => {
        jest.resetAllMocks();
    });

    it("should issue command", async () => {
        IssueTso.issueTsoCmd = jest.fn((session, acc, cmd, prof) => {
            expect(prof).toBeDefined();
            expect(prof).toMatchSnapshot();
            return StartTsoData.SAMPLE_ISSUE_RESPONSE_WITH_MSG;
        });
        const handler = new Command.default();
        const params = Object.assign({}, ...[DEFAULT_PARAMETERS]);
        params.arguments.acc = "acc";
        params.arguments.cmd = "time";
        await handler.process(params);
        expect(IssueTso.issueTsoCmd).toHaveBeenCalledTimes(1);
    });

    it("should be able respond with error message", async () => {
        const failMessage =
            'IZUG1126E: z/OSMF cannot correlate the request for key "ZOSMFAD-SYS2-55-aaakaaac"\n' +
            "with an active z/OS application session.";
        let error;
        IssueTso.issueTsoCmd = jest.fn((session, servletKey) => {
            throw new ImperativeError({ msg: failMessage });
        });
        const handler = new Command.default();
        const params = Object.assign({}, ...[DEFAULT_PARAMETERS]);
        params.arguments.servletKey = "ZOSMFAD-SYS2-55-aaakaaac";
        params.arguments.servletKey = "data";
        try {
            await handler.process(params);
        } catch (thrownError) {
            error = thrownError;
        }
        expect(IssueTso.issueTsoCmd).toHaveBeenCalledTimes(1);
        expect(error).toBeDefined();
        expect(error instanceof ImperativeError).toBe(true);
        expect(error.message).toMatchSnapshot();
    });

    it("should issue command with issueTsoCmd", async () => {
        IssueTso.issueTsoCmd = jest.fn().mockReturnValue({
            success: true,
            startReady: true,
            zosmfResponse: {
                cmdResponse: [
                    {
                        message:
                            "IKJ56650I TIME-00:00:00 AM. CPU-00:00:00 SERVICE-554 SESSION-00:00:00 JANUARY 1,2024",
                    },
                    {
                        message: "READY ",
                    },
                ],
                tsoPromptReceived: "Y",
            },
            commandResponse:
                "IKJ56650I TIME-00:00:00 AM. CPU-00:00:00 SERVICE-554 SESSION-00:00:00 JANUARY 1,2024\nREADY ",
        });
        const handler = new Command.default();
        const params = Object.assign({}, ...[PARAMS_NO_SSM_STATEFUL]);
        await handler.process(params);
        expect(IssueTso.issueTsoCmd).toHaveBeenCalledTimes(1);
    });
});
