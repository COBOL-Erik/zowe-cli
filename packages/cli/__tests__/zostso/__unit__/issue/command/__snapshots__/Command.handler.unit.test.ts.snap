// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue command handler tests should be able respond with error message 1`] = `
"IZUG1126E: z/OSMF cannot correlate the request for key \\"ZOSMFAD-SYS2-55-aaakaaac\\"
with an active z/OS application session."
`;

exports[`issue command handler tests should issue command 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Session {
        "mISession": Object {
          "base64EncodedAuth": "c29tZW9uZTpmYWtl",
          "basePath": "",
          "hostname": "somewhere.com",
          "password": "fake",
          "port": "43443",
          "protocol": "https",
          "rejectUnauthorized": true,
          "secureProtocol": "SSLv23_method",
          "strictSSL": true,
          "type": "basic",
          "user": "someone",
        },
        "mLog": Logger {
          "category": "imperative",
          "initStatus": false,
          "mJsLogger": Logger {
            "callStackSkipIndex": 0,
            "category": "imperative",
            "context": Object {},
            "parseCallStack": [Function],
          },
        },
      },
      undefined,
      Object {
        "isStateful": undefined,
        "suppressStartupMessage": undefined,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "incomplete",
      "value": undefined,
    },
  ],
}
`;

exports[`issue command handler tests should issue command 2`] = `"messages from TSO"`;

exports[`issue command handler tests should issue command 3`] = `"this is the command response"`;

exports[`issue command handler tests should issue command 4`] = `
Object {
  "commandResponse": "this is the command response",
  "servletKey": "ZOSMFAD-SYS2-55-aaakaaac",
  "startResponse": Object {
    "messages": "messages from TSO",
  },
  "success": true,
  "zosmfPingResponse": Object {
    "reused": false,
    "servletKey": "ZOSMFAD-SYS2-55-aaakaaac",
    "timeout": false,
    "ver": "0100",
  },
}
`;
